src/main/java/seedu/address/ui/StudentCard.java

```diff
@@ -60,9 +64,21 @@ public StudentCard(Student student, int displayedIndex) {
         student.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
-        student.getLabs()
-                .asUnmodifiableObservableList()
-                .forEach(lab -> labCard.getChildren().add(new LabLabel(lab)));
+        initLabCard(student);
+    }
+
+    /**
+     * Initializes the ListView {@code labCard}.
+     *
+     * @param student The Student object that this StudentCard belongs to.
+     */
+    private void initLabCard(Student student) {
+        // labCard related settings
+        labCard.setItems(student.getLabs().asUnmodifiableObservableList());
+        labCard.setCellFactory(listView -> new LabListViewCell());
+        labCard.setOrientation(Orientation.HORIZONTAL);
+        // this is hardcoded, might need to be changed.
+        labCard.setMaxHeight(30);
```

Comment:
I think hard coding it is okay? We don't really want the ListView to extend vertically if we expand our window right?